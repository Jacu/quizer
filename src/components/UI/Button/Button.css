.Button{
    color: inherit;
    text-transform: uppercase;
    text-decoration: none;
    cursor: pointer;
    background: white;
    padding: 20px 50px;    
    border-radius: 25px;
    font-weight: bold;
    font-size: 2rem;
    box-shadow: 2px 2px 10px 2px rgba(0,0,0,0.25);
    transition: transform 0.25s;
    margin: 2rem;
}

.Button:hover{
    transform: scale(1.2,1.2);    
}