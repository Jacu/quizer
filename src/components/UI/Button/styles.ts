import styled from "styled-components";
import { PRIMARY_THEME } from "~/utils/colors";

interface IButtonStyle {
    size: number,
    uppercase?: boolean,
    bold?: boolean,
}

export const Button = styled.div<IButtonStyle>`
    color: white;
    text-decoration: none;
    text-transform: ${props => props.uppercase ? 'uppercase' : 'none'};
    cursor: pointer;
    background: ${PRIMARY_THEME.BASE};
    padding: 15px 25px;
    font-size: ${props => `${props.size}rem`};
    font-weight: ${props => props.bold ? 'bold' : 'normal' };
    margin: 2rem;
    margin-bottom: 90px;
    :hover {
        background: ${PRIMARY_THEME.DARK};
    }
`;