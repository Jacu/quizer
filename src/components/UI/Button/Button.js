import React from 'react';
import { Link } from 'react-router-dom';
import styles from './Button.css';

const Button = props => {
    return (
        <Link
            className={styles.Button}
            to={props.link}
            onClick={props.onClick}>
            {props.label}
        </Link>
    )
}

export default Button;