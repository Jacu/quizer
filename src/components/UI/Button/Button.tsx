import React from 'react';
import * as styled from './styles';

interface ButtonProps {
    link: string,
    label: string,
    onClick: () => void,
};

const Button: React.FC<ButtonProps> = props => {
    return (
        <styled.Button
            children={props.label}
            to={props.link}
            onClick={props.onClick} />
    )
}

export default Button;