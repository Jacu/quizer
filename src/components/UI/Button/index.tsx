import React from 'react';
import * as styled from './styles';

interface IButton {
    label?: string,
    size?: number,
    uppercase?: boolean,
    bold?: boolean,
    onClick: () => void,
}

const Button: React.FC<IButton> = ({label='', onClick, size=1.25, uppercase, bold}) => {
    return (
        <styled.Button
            bold={bold}
            uppercase={uppercase}
            size={size}
            children={label}
            onClick={onClick} />
    )
}

export default Button;