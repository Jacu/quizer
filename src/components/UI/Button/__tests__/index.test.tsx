import React from 'react';
import Button from '../';
import { render, fireEvent, screen } from '@testing-library/react';
import '@testing-library/jest-dom';

const onClick = jest.fn();
const props = {
    onClick,
}

describe('Button', () => {
    it('should render with default props', () => {
        const { asFragment } = render(<Button {...props} />);
        expect(asFragment()).toMatchSnapshot();
    });

    it('should display label', () => {
        const label = 'test label';
        const { container } = render(<Button {...props} label={label} />);
        expect(screen.getAllByText(label)).toHaveLength(1);
        expect(container.firstChild).toHaveTextContent(label);
    });

    it('should handle click', () => {
        const label = 'test label';
        const { getByText } = render(<Button {...props} label={label} />);
        fireEvent.click(getByText(label));
        expect(onClick).toHaveBeenCalled();
    });

    it('should uppercase label', () => {
        const label = 'test label';
        const { getByText } = render(<Button {...props} label={label} uppercase />);
        expect(getByText(label)).toHaveStyle(` text-transform: uppercase; `);
    });

    it('should use provided font-size for label', () => {
        const size = 5;
        const label = 'test label';
        render(<Button {...props} label={label} size={size} />);
        expect(screen.getByText(label)).toHaveStyle(` font-size: ${size}rem; `);
    });

    it('should use default font-size for label', () => {
        const label = 'test label';
        render(<Button {...props} label={label} />);
        expect(screen.getByText(label)).toHaveStyle(` font-size: 1.25rem; `);
    });

    it('should bold label', () => {
        const label = 'test label';
        render(<Button {...props} label={label} bold />);
        expect(screen.getByText(label)).toHaveStyle(` font-weight: bold; `);
    });
});