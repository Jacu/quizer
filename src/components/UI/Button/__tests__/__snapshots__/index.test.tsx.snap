// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should render with default props 1`] = `
<DocumentFragment>
  <div
    class="sc-AxjAm tsLrJ"
    size="1.25"
  />
</DocumentFragment>
`;
